### PR TITLE
Add optional documentation link to help text

### PR DESCRIPTION
**Pull request by akshaydewan**
Related to #6362 

Description: A lot of fields in the pipeline config page have a link to more documentation. This adds an optional property called `docLink` to the help text.

Example:
Screenshot 2019-08-26 at 17 51 16
`More...` opens the linked gocd documentation in a new tab

### CODE REVIEW

**Comment by GaneshSPatil** [comment]
@akshaydewan -- The "more" text initially gave me an impression that more information will be rendered inline. (similar to "show more.." and "show less..")

Can we change the wording to "Learn more"?

So:
- "more.."/"less.." - shows inline information.
- "learn more" - links to the external documentation.

Or is there a better way to reduce the confusion between inline help text vs external links?

**Comment by GaneshSPatil** [comment]
\cc: @streisguth  @marques-work -- WDYT?

**Comment by akshaydewan** [comment]
"Learn more..." seems ok to me.

File: server/webapp/WEB-INF/rails/webpack/views/components/forms/input_fields.tsx [hunk 1]
```diff
@@ -202,9 +205,16 @@ class HelpText extends MithrilViewComponent<HelpTextComponentAttrs> {
     }
   }
 
+  static maybeDocLink(docLink?: string) {
+    if (!_.isEmpty(docLink)) {
+      return <span>&nbsp;<Link href={docsUrl(docLink)} target="_blank">More&#8230;</Link></span>;
```
Comment (ketan) [inline server/webapp/WEB-INF/rails/webpack/views/components/forms/input_fields.tsx]: See https://www.owasp.org/index.php/Reverse_Tabnabbing. Perhaps make the `Link` component smart enough, based on the `href` attribute to add additional attrs?

Comment (akshaydewan) [reply server/webapp/WEB-INF/rails/webpack/views/components/forms/input_fields.tsx]: Thanks for pointing that out. About the `Link` component, do you mean checking if the link specifies `target` and then adding `rel="noopener noreferrer"` if needed?

**Comment by akshaydewan** [comment]
I've made some changes 
- Added the `rel="noopener noreferrer"` when a `target` is specified
- Changed the text to "Learn More"
- Used the external link icon instead of ellipsis
Screenshot 2019-08-27 at 11 53 07

File: server/webapp/WEB-INF/rails/webpack/views/components/forms/input_fields.tsx (hunk 1 above)
Comment (ketan) [reply server/webapp/WEB-INF/rails/webpack/views/components/forms/input_fields.tsx]: What I meant is to make the `Link` component smart enough to specifically detect external links (`http(s)://`) and, based on that, automatically add the `target="_blank"` and `rel="noopener noreferrer"` attributes.

**Pull request merged**